fordel-web/src/main/java/no/nav/foreldrepenger/fordel/web/app/selftest/HealthCheckRestService.java

```diff
@@ -1,81 +1,77 @@
-
     private static final String RESPONSE_CACHE_KEY = "Cache-Control";
     private static final String RESPONSE_CACHE_VAL = "must-revalidate,no-cache,no-store";
     private static final String RESPONSE_OK = "OK";
 
-    private ApplicationServiceStarter starterService;
-    private DatabaseHealthCheck databaseHealthCheck;
+    private List<LivenessAware> live;
+    private List<ReadinessAware> ready;
 
     public HealthCheckRestService() {
         // CDI
     }
 
     @Inject
-    public HealthCheckRestService(ApplicationServiceStarter starterService, DatabaseHealthCheck databaseHealthCheck) {
-        this.starterService = starterService;
-        this.databaseHealthCheck = databaseHealthCheck;
+    public HealthCheckRestService(@Any Instance<LivenessAware> livenessAware, @Any Instance<ReadinessAware> readinessAware) {
+        this(livenessAware.stream().collect(toList()), readinessAware.stream().collect(toList()));
+    }
+
+    public HealthCheckRestService(List<LivenessAware> live, List<ReadinessAware> ready) {
+        this.live = live;
+        this.ready = ready;
     }
 
     @GET
     @Path("isAlive")
     @Operation(description = "sjekker om poden lever", tags = "nais", hidden = true)
     public Response isAlive() {
-        if (starterService.isKafkaAlive()) {
+        if (live.stream().allMatch(LivenessAware::isAlive)) {
             return Response
                     .ok(RESPONSE_OK)
                     .header(RESPONSE_CACHE_KEY, RESPONSE_CACHE_VAL)
                     .build();
-        } else {
-            return Response
-                    .serverError()
-                    .header(RESPONSE_CACHE_KEY, RESPONSE_CACHE_VAL)
-                    .build();
         }
+        return Response
+                .serverError()
+                .header(RESPONSE_CACHE_KEY, RESPONSE_CACHE_VAL)
+                .build();
+
     }
 
     @GET
     @Path("isReady")
     @Operation(description = "sjekker om poden er klar", tags = "nais", hidden = true)
     public Response isReady() {
-        if (starterService.isKafkaAlive() && databaseHealthCheck.isReady()) {
+        if (ready.stream().allMatch(ReadinessAware::isReady)) {
             return Response
                     .ok(RESPONSE_OK)
                     .header(RESPONSE_CACHE_KEY, RESPONSE_CACHE_VAL)
                     .build();
-        } else {
-            return Response
-                    .status(Response.Status.SERVICE_UNAVAILABLE)
-                    .header(RESPONSE_CACHE_KEY, RESPONSE_CACHE_VAL)
-                    .build();
         }
+        return Response
+                .status(Response.Status.SERVICE_UNAVAILABLE)
+                .header(RESPONSE_CACHE_KEY, RESPONSE_CACHE_VAL)
+                .build();
     }
 
-    @GET
-    @Path("preStop")
-    @Operation(description = "kalles på før stopp", tags = "nais", hidden = true)
-    public Response preStop() {
```

Comment:
Denne her kalles av nais før containeren blir stoppet og er ansvarlig for "gentle stop". Tror nais kaller på dette endepunktet og så venter i 30 sek før den terminerer poden https://github.com/navikt/fpfordel/blob/master/.deploy/naiserator.yaml#L26

Reply:
lagt tilbake